git-lfs.go

```diff
@@ -1,11 +1,37 @@
 package main
 
 import (
+	"fmt"
+	"os"
+	"os/signal"
+	"sync"
+
 	"github.com/github/git-lfs/commands"
 	"github.com/github/git-lfs/lfs"
 )
 
 func main() {
+	c := make(chan os.Signal)
+	signal.Notify(c)
+
+	var once sync.Once
+
+	go func() {
+		for {
+			sig := <-c
+			once.Do(lfs.ClearTempObjects)
```

Comment:
Hmm, seems Go only exposes SIGINT and SIGKILL for filtering. I think if you want to handle more types you have to use the syscall package which will be less portable - you'd also have to make the signal channel large enough to handle the non-blocking calls for other signals. Maybe it's best just to handle those 2 in the Notify().